packages/cli/src/lib/setup/setupUpload.ts

```diff
@@ -530,7 +530,8 @@ export class Upload {
 
         if (!fs.existsSync(dir)) {
             // www folder have not all adapters. So show warning only for admin folder
-            (isAdmin || (cfg && cfg.common && cfg.common.onlyWWW)) &&
+            // widgets does not have www folder, but they onlyWWW flag
+            (isAdmin || (cfg?.common?.onlyWWW && !cfg?.common?.visWidgets)) &&
```

Comment:
```suggestion
            (isAdmin || (cfg?.common?.onlyWWW && !cfg.common.visWidgets)) &&
```
Still prefer this one, `cfg.common` always exists on entering the `&&` and it's hard to read at least in my mind beacause basically a non existing `cfg` would evaluate to `true`, even if this cannot happen in this scenario it is weird to read for me.